### PR TITLE
Dialogs variant consistently

### PR DESCRIPTION
**Pull request by jelly**
Go smoll

image

### CODE REVIEW

**Comment by allisonkarlitskaya** [comment]
Can you give before & after screenshots?  I think we might actually want small here — that's what I ended up using for the properties (eg. permissions) dialog.

**Comment by jelly** [comment]
> Can you give before & after screenshots? I think we might actually want small here — that's what I ended up using for the properties (eg. permissions) dialog.

We don't want what it was for sure that takes up the full width however, I see we now have small and medium mixed.

**Comment by marusak** [comment]
I did some digging and we use in cockpit 11 modals with `small`, around 30 with `medium` and no `large`. In podman we always use `medium`. In machines we use 17 modal with `medium` and 11 with `small`.

We also have `pkg/lib/cockpit-components-dialog.jsx:            <Modal position="top" variant={this.props.variant || "medium"}` -  so we default to medium.

So I thing we should be using `medium` by default. Some of the `small` ones might be just wrong usage but maybe some of them are there for a good reason. We can verify with @garrett or just go with medium everywhere.

**Review by garrett — CHANGES_REQUESTED** [review]
So I think this PR is mostly correct. However, we probably do need a longer width for the create link modal (medium), as it may require a full path. The others should be small.

---

General Cockpit unwritten* rule for modal sizing:

We should generally go with small, unless the forms require more width, then we go with medium. And if they need large, then they most likely need a redesign as they're probably too complex. (There can be exceptions.)

Reasoning: It's easier to read narrower text and elements. Generally, [optimal line width is between 50 to 75 characters](https://baymard.com/blog/line-length-readability). The small modal width is roughly 70 characters wide. (It's sometimes a little longer and sometimes shorter, depending on word length.)

However, when you do need to enter longer content, such as full paths, you need extra space, so sometimes we need to go longer and bump up to medium size.

(* I just wrote it here, so I guess it's a written rule now? :wink:)

File: src/fileActions.jsx [hunk 1]
```diff
@@ -235,6 +235,7 @@ export const CreateDirectoryModal = ({ selected, currentPath }) => {
           title={_("Create directory")}
           isOpen
           onClose={Dialogs.close}
+          variant={ModalVariant.medium}
```
Comment (KKoukiou) [inline src/fileActions.jsx:239]: You say small in the commit message but use medium

**Comment by allisonkarlitskaya** [comment]
I think this is @garrett's to review.

**Comment by garrett** [comment]
Why was create directory changed to medium and the create link left as small? Shouldn't it be the opposite? Create directory is contextual to your current location and shouldn't be long (unless it accepts sub-directories and does a `mkdir -p`), whereas link could be absolute and therefore could be long.

**Comment by jelly** [comment]
> Why was create directory changed to medium and the create link left as small? Shouldn't it be the opposite? Create directory is contextual to your current location and shouldn't be long (unless it accepts sub-directories and does a `mkdir -p`), whereas link could be absolute and therefore could be long.

Ok, in that case rename should also be small.

**Review by garrett — APPROVED** [review]
Thanks!

File: src/fileActions.jsx [hunk 2]
```diff
@@ -140,7 +140,7 @@ export const ConfirmDeletionDialog = ({
           position="top"
           title={modalTitle}
           titleIconVariant="warning"
-          variant={ModalVariant.small}
+          variant={ModalVariant.medum}
```
Comment (martinpitt) [inline src/fileActions.jsx:143]: as is this is `undefined`, it means that it's happy with the default value?

**Review by martinpitt — CHANGES_REQUESTED** [review]

**Review by martinpitt — APPROVED** [review]

**Pull request merged**